src/Hint/Context.hs

```diff
@@ -385,14 +380,27 @@ reset = do -- clean up context
            cleanPhantomModules
            --
            -- Now, install a support module
-           installSupportModule
+           res <- installSupportModule Nothing
+           mayFail (return $ guard (isSucceeded res) >> Just ())
 
 -- Load a phantom module with all the symbols from the prelude we need
-installSupportModule :: MonadInterpreter m => m ()
-installSupportModule = do mod <- addPhantomModule support_module
-                          onState (\st -> st{hintSupportModule = mod})
-                          mod' <- findModule (pmName mod)
-                          runGhc $ setContext [mod'] []
+installSupportModule :: MonadInterpreter m => Maybe [GHC.Target] -> m GHC.SuccessFlag
+installSupportModule tM = do mod <- addPhantomModule support_module
+                             onState (\st -> st{hintSupportModule = mod})
+                             case tM of
+                                Nothing -> do
+                                  mod' <- findModule (pmName mod)
+                                  runGhc $ setContext [mod'] []
+                                  return GHC.Succeeded
+                                Just ts -> do
+                                  runGhc $ GHC.setTargets ts
+                                  df <- runGhc GHC.getSessionDynFlags
+                                  let t = GHC.fileTarget df (pmFile mod)
+                                  runGhc $ GHC.addTarget t
+                                  res <- runGhc $ GHC.load GHC.LoadAllTargets
```

Comment:
I am confused. `addPhantomModule` is basically running those two lines, `addTarget` and then `LoadAllTargets`. So why do we need to reset the targets and then run those two lines again after `addPhantomModule` returns? Could we maybe call `setTargets` before calling `addPhantomModule` instead?

Reply:
567c057bbb89f5b694ce0a0881ee2b77b104317c now calls `setTargets` before calling `addPhantomModule`